File: frontend/src/api/file.js
/*
 * @Author: iRorikon
 * @Date: 2023-04-06 20:00:21
 * @FilePath: \http-file\frontend\src\api\file.js
 */
import request from '~/utils/request'

export const getFileinfo = data => {
  return request({
    url: 'file/info',
    method: 'post',
    data
  })
}

export const getFileList = data => {
  return request({
    url: 'file/list',
    method: 'post',
    data
  })
}

export const renameFile = data => {
  return request({
    url: 'file/copy',
    method: 'post',
    data
  })
}

export const removeFile = data => {
  return request({
    url: 'file/remove',
    method: 'post',
    data
  })
}

export const copyFile = data => {
  return request({
    url: 'file/copy',
    method: 'post',
    data
  })
}

export const moveFile = data => {
  return request({
    url: 'file/move',
    method: 'post',
    data
  })
}

export const searchFile = data => {
  return request({
    url: 'file/search',
    method: 'post',
    data
  })
}

File: backend/router/storage/file.go
/*
 * @Author: iRorikon
 * @Date: 2023-04-07 22:56:25
 * @FilePath: \http-file\backend\router\storage\file.go
 */
package storage

import (
	"github.com/gin-gonic/gin"
	"github.com/irorikon/http-file/controller"
)

type FileRouter struct{}

func (fr *FileRouter) InitFileRouter(Router *gin.RouterGroup) {
	fileRouter := Router.Group("file")
	fileController := controller.ControllerAPP.File.FileController
	{
		fileRouter.POST("list", fileController.GetFileList)
		fileRouter.POST("info", fileController.GetFileInfo)
		fileRouter.POST("mkdir", fileController.MakeDir)
		fileRouter.POST("copy", fileController.Cpoy)
		fileRouter.POST("move", fileController.Move)
		fileRouter.POST("remove", fileController.Remove)
		fileRouter.POST("search", fileController.Search)
		fileRouter.GET("d/*link", fileController.Download)
	}
}
